sdk/examples/xo_python/sawtooth_xo/processor/xo_state.py

```diff
@@ -0,0 +1,150 @@
+
+
+def _make_xo_address(name):
+    return XO_NAMESPACE + \
+        hashlib.sha512(name.encode('utf-8')).hexdigest()[:64]
+
+
+class Game(object):
+    def __init__(self, name, board, state, player1, player2):
+        self.name = name
+        self.board = board
+        self.state = state
+        self.player1 = player1
+        self.player2 = player2
+
+
+class XoState(object):
+
+    TIMEOUT = 3
+
+    def __init__(self, context, game_name):
+        """Constructor.
+
+        Args:
+            context (sawtooth_sdk.processor.context.Context): Access to
+                validator state from within the transaction processor.
+
+            game_name (str): The name of the game currently being acted on.
+        """
+
+        self._context = context
+        self._game_name = game_name
+        self._game_address = _make_xo_address(game_name)
+
+        state_entries = context.get_state(
+            [self._game_address],
+            timeout=self.TIMEOUT)
+
+        # context.get_state() returns a list. If no data has been stored yet
+        # at the given address, it will be empty.
+        if state_entries:
+            try:
+                state_data = state_entries[0].data
+
+                games = {
+                    name: (board, state, player1, player2)
+                    for name, board, state, player1, player2 in [
+                        game.split(',')
+                        for game in state_data.decode().split('|')
+                    ]
+                }
+
+                board, state, player1, player2 = \
+                    games[self._game_name]
+
+            except ValueError:
+                raise InternalError("Failed to deserialize game data.")
+
+            game = Game(name=game_name,
+                        board=board,
+                        state=state,
+                        player1=player1,
+                        player2=player2)
+
+            self._games = games
+            self._games[self._game_name] = game
+        else:
+            self._games = {}
+
+    def delete_game(self, game_name):
+        """Delete the Game named game_name from state.
+
+        Args:
+            game_name (str): The name.
+        """
+
+        del self._games[game_name]
+        if self._games:
+            self._store_game()
+        else:
+            self._delete_game()
+
+    def set_game(self, game):
+        """Store the game in the validator state.
+
+        Args:
+            game (Game): The information specifying the current game.
+        """
+
+        self._games[game.name] = game
+
+        self._store_game()
+
+    def get_game(self, game_name):
+        """Get the game associated with game_name.
+
+        Args:
+            game_name (str): The name.
+
+        Returns:
+            (Game): All the information specifying a game.
+        """
+        try:
+            return self._games[game_name]
+        except KeyError:
+            return None
+
+    def _store_game(self):
```

Comment:
This is too confusing for example code.  Maybe we can break seralization/deserializion of the game state out to separate functions.